### PR TITLE
[Stack-hci] `az stack-hci arc-setting`: Add new command `initialize-disable-proces` to support initializing disable proces

### PR DESCRIPTION
**Pull request by Jing-song**
* Add new command `az stack-hci arc-setting consent-and-install-default-extension` to support consenting and installing default extension for arc settings
* Add new command `az stack-hci arc-setting initialize-disable-proces` to support initializing and disabling proces
* Add new command `az stack-hci cluster extend-software-assurance-benefit` to support extending software assurance benefit
* Add new subcommand `az stack-hci cluster identity assign/remove` to support managing cluster identity
* Fix the conflict key for `az stack-hci cluster list` and `az stack-hci cluster show`

close https://github.com/Azure/azure-cli/issues/26411

AAZ https://github.com/Azure/aaz/pull/200

### CODE REVIEW

**Comment by yonzhan** [comment]
Stack-hci

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_consent_and_install_default_extension.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+# --------------------------------------------------------------------------------------------
+
+# pylint: skip-file
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "stack-hci arc-setting consent-and-install-default-extension",
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_consent_and_install_default_extension.py:15]: This command signature is too long. Could we simplify it to `stack-hci arc-setting install-default-extension`? @priyjain358 What do you think of it?
```suggestion
    "stack-hci arc-setting install-default-extension",
```

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_initialize_disable_proces.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "stack-hci arc-setting initialize-disable-proces",
+)
+class InitializeDisableProces(AAZCommand):
+    """Initializes ARC Disable process on the cluster
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_initialize_disable_proces.py:18]:
```suggestion
    """Initialize ARC Disable process on the cluster
```

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/cluster/_create.py [hunk 3]
```diff
@@ -75,6 +75,24 @@ def _build_arguments_schema(cls, *args, **kwargs):
...
 
+        _args_schema = cls._args_schema
+        _args_schema.type = AAZStrArg(
+            options=["--type"],
+            arg_group="Identity",
+            help="Type of managed service identity (where both SystemAssigned and UserAssigned types are allowed).",
+            enum={"None": "None", "SystemAssigned": "SystemAssigned", "SystemAssigned, UserAssigned": "SystemAssigned, UserAssigned", "UserAssigned": "UserAssigned"},
+        )
+        _args_schema.user_assigned_identities = AAZDictArg(
+            options=["--user-assigned-identities"],
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/cluster/_create.py:86]: Could you please follow this guideline [managed_identity_command_guideline.md](https://github.com/Azure/azure-cli/blob/dev/doc/managed_identity_command_guideline.md) to design the managed identity related parameters?

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/cluster/_extend_software_assurance_benefit.py [hunk 4]
```diff
@@ -0,0 +1,468 @@
...
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "stack-hci cluster extend-software-assurance-benefit",
+)
+class ExtendSoftwareAssuranceBenefit(AAZCommand):
+    """Extends Software Assurance Benefit to a cluster
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/cluster/_extend_software_assurance_benefit.py:18]:
```suggestion
    """Extend software assurance benefit to a cluster
```

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/cluster/_update.py [hunk 5]
```diff
@@ -73,6 +73,26 @@ def _build_arguments_schema(cls, *args, **kwargs):
...
 
+        _args_schema = cls._args_schema
+        _args_schema.type = AAZStrArg(
+            options=["--type"],
+            arg_group="Identity",
+            help="Type of managed service identity (where both SystemAssigned and UserAssigned types are allowed).",
+            enum={"None": "None", "SystemAssigned": "SystemAssigned", "SystemAssigned, UserAssigned": "SystemAssigned, UserAssigned", "UserAssigned": "UserAssigned"},
+        )
+        _args_schema.user_assigned_identities = AAZDictArg(
+            options=["--user-assigned-identities"],
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/cluster/_update.py:84]: Same as above

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_consent_and_install_default_extension.py (hunk 1 above)
Comment (priyjain358) [reply src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_consent_and_install_default_extension.py:15]: @zhoxing-ms we can't change this command signature.

Comment (zhoxing-ms) [reply src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_consent_and_install_default_extension.py:15]: @priyjain358 May I ask what is the reason?

File: src/stack-hci/HISTORY.rst [hunk 6]
```diff
@@ -3,6 +3,12 @@
 Release History
 ===============
 
+0.1.8
+++++++
+* Add new command `consent-and-install-default-extension` `initialize-disable-proces` for `az stack-hci arc-setting`
+* Add new command `extend-software-assurance-benefit` for `az stack-hci cluster`
```
Comment (zhoxing-ms) [inline src/stack-hci/HISTORY.rst:9]:
```suggestion
* Add new command `az stack-hci arc-setting consent-and-install-default-extension` to support xxx
* Add new command `az stack-hci arc-setting initialize-disable-proces` to support xxx
* Add new command `az stack-hci cluster extend-software-assurance-benefit` to support xxx
```

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/cluster/_create.py [hunk 7]
```diff
@@ -70,11 +75,26 @@ def _build_arguments_schema(cls, *args, **kwargs):
...
+        identity.type = AAZStrArg(
+            options=["type"],
+            help="Type of managed service identity (where both SystemAssigned and UserAssigned types are allowed).",
+            required=True,
+            enum={"None": "None", "SystemAssigned": "SystemAssigned", "SystemAssigned, UserAssigned": "SystemAssigned, UserAssigned", "UserAssigned": "UserAssigned"},
+        )
+        identity.user_assigned_identities = AAZDictArg(
+            options=["user-assigned-identities"],
+            help="The set of user assigned identities associated with the resource. The userAssignedIdentities dictionary keys will be ARM resource ids in the form: '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}. The dictionary values can be empty objects ({}) in requests.",
+        )
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/cluster/_create.py:88]: Could we remove these parameters which have been replaced by handwritten parameters?

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/cluster/_create.py [hunk 8]
```diff
@@ -56,6 +56,11 @@ def _build_arguments_schema(cls, *args, **kwargs):
         # define Arg Group "Cluster"
 
         _args_schema = cls._args_schema
+        _args_schema.identity = AAZObjectArg(
+            options=["--identity"],
+            arg_group="Cluster",
+            help="Identity of Cluster resource",
+        )
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/cluster/_create.py:63]: Could we remove these parameters which have been replaced by handwritten parameters?

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/cluster/_update.py [hunk 9]
```diff
@@ -59,20 +59,42 @@ def _build_arguments_schema(cls, *args, **kwargs):
...
+        identity.user_assigned_identities = AAZDictArg(
+            options=["user-assigned-identities"],
+            help="The set of user assigned identities associated with the resource. The userAssignedIdentities dictionary keys will be ARM resource ids in the form: '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}. The dictionary values can be empty objects ({}) in requests.",
+            nullable=True,
+        )
+
+        user_assigned_identities = cls._args_schema.identity.user_assigned_identities
+        user_assigned_identities.Element = AAZObjectArg(
+            nullable=True,
+            blank={},
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/cluster/_update.py:90]: Same as above. Please note this is a update related scenario, we need to follow this [guideline](https://github.com/Azure/azure-cli/blob/dev/doc/managed_identity_command_guideline.md#operate-managed-identity-on-existing-resource) to create a `az <resource> identity assign` sub-command.

File: src/stack-hci/azext_stack_hci/custom.py [hunk 10]
```diff
@@ -6,13 +6,84 @@
...
+        args_schema.mi_user_assigned = AAZListArg(
+            options=["--mi-user-assigned"],
+            help="Space separated resource IDs to add user-assigned identities.",
+        )
+        args_schema.mi_user_assigned.Element = AAZResourceIdArg(
+            fmt=AAZResourceIdArgFormat(template="/subscriptions/{subscription}/resourceGroups/{resource_group}"
+                                                "/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{}")
+        )
+        args_schema.identity._registered = False
+        return args_schema
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/custom.py:75]: Same as above

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/cluster/_create.py (hunk 8 above)
Comment (Jing-song) [reply src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/cluster/_create.py:63]: `args_schema.identity._registered = False` 
the parameter has been hidden, it is still used elsewhere,so it cannot be removed

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_consent_and_install_default_extension.py (hunk 1 above)
Comment (priyjain358) [reply src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_consent_and_install_default_extension.py:15]: just install-default-extension doesn't capture the whole intent behind the function.

Comment (zhoxing-ms) [reply src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_consent_and_install_default_extension.py:15]: OK, got it

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/extension/_create.py [hunk 11]
```diff
@@ -92,11 +92,6 @@ def _build_arguments_schema(cls, *args, **kwargs):
             arg_group="ExtensionParameters",
             help="Json formatted public settings for the extension.",
         )
-        _args_schema.type = AAZStrArg(
-            options=["--type"],
-            arg_group="ExtensionParameters",
-            help="Specifies the type of the extension; an example is \"CustomScriptExtension\".",
-        )
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/extension/_create.py:99]: Why do we need to remove it?

File: src/stack-hci/azext_stack_hci/tests/latest/test_stack_hci_commands.py [hunk 12]
```diff
@@ -107,3 +116,29 @@ def test_stack_hci_extension_crud(self):
...
+        self.kwargs['client_id'] = self.cmd('ad app create --display-name {app_name}').get_output_in_json()['appId']
+        self.kwargs['tenant_id'] = self.cmd('account show').get_output_in_json()['tenantId']
+
+        self.cmd('stack-hci cluster create -n {cluster_name} -g {rg} --aad-client-id {client_id} --aad-tenant-id {tenant_id} --tags key0=value0 --mi-system-assigned', checks=[
+            self.check('identity.type', 'SystemAssigned')
+        ])
+        try:
+            self.cmd('stack-hci cluster identity assign --cluster-name {cluster_name} -g {rg} --user-assigned id1')
+        except Exception as ex:
+            self.assertTrue("Resource type 'Microsoft.AzureStackHCI/clusters' does not support creation of 'UserAssigned' resource identity." in str(ex))
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/tests/latest/test_stack_hci_commands.py:136]: I would like to confirm with you that `StackHCI` currently does not support user assigned identity, which is in line with expectations? If so, can we not add user assigned identity related parameters for now?

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**